### PR TITLE
feat: upgrade nix version

### PR DESCRIPTION
**Pull request by thaodt**
The original ipipe depended on the old nix version which caused a vulnerability on [our pipeline audit check](https://github.com/LitheumOrg/LitheumCore/runs/6145061721?check_suite_focus=true#step:5:18)

This PR served for upgrading nix version.
But this version won't have these error types anymore `nix::Error::InvalidPath`, `nix::Error::InvalidUtf8`  & `nix::Error::UnsupportedOperation`, for further details please check [this discussion](https://github.com/nix-rust/nix/issues/613).
TLDR: No conversion between `Errno` and `std::io::Error` anymore.
My changes are to adapt this, but im not sure if its enough. 
Please do a rough review. Thanks.

### CODE REVIEW

**Review by clayrab — APPROVED** [review]
LGTM, just remove the // code
